mantidproject/mantid
# Stop masked data in MDWorkspaces being displayed in slice viewer

Closes #14523 

Masked data in MDWorkspaces now is now displayed with a value of 0 in the Slice Viewer. This is consistent with MatrixWorkspaces.

**To Test:**
Use the following script to make a masked and unmasked workspace of each MDWorkspace type (MDHistoWorkspace and MDEventWorkspace).
Compare the appearance of masked and unmasked of each type in the Slice Viewer. The central unit cube of the masked workspaces should be masked and data cursor should show a value of 0 ("Intensity=" in the top right of the viewer window).

``` python

ws_mdhw = CreateMDWorkspace(Dimensions='3', Extents='-2,2,-2,2,-2,2', Names='h,k,l',
                       Units='rlu,rlu,rlu', SplitInto='12')
FakeMDEventData(ws_mdhw, UniformParams='1000', RandomizeSignal='1')
ws_mdhw = BinMD(ws_mdhw,AlignedDim0='h,-2,2,12',AlignedDim1='k,-2,2,12',AlignedDim2='l,-2,2,12')
ws_masked_mdhw = CloneMDWorkspace(ws_mdhw)
MaskMD(Workspace=ws_masked_mdhw,Dimensions="h,k,l",Extents="-1,1,-1,1,-1,1")

ws_mdew = CreateMDWorkspace(Dimensions='3', Extents='-2,2,-2,2,-2,2', Names='h,k,l',
                       Units='rlu,rlu,rlu', SplitInto='12')
FakeMDEventData(ws_mdew, UniformParams='1000', RandomizeSignal='1')
ws_masked_mdew = CloneMDWorkspace(ws_mdew)
MaskMD(Workspace=ws_masked_mdew,Dimensions="h,k,l",Extents="-1,1,-1,1,-1,1")

```
